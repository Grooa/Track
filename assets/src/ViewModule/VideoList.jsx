import PropTypes from 'prop-types';

import VideoCard from './VideoCard';

const VideoList = ({ videos }) => <ul className="course-list">
  {videos.map((v, index) => <li key={`video-${index}`}>
    <VideoCard video={v} number={index + 1}/>
  </li>)}
</ul>;

VideoList.propTypes = {
  videos: PropTypes.array.isRequired,
};

export default VideoList;
