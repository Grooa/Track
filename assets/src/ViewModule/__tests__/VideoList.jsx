import { render } from 'enzyme';

import VideoList from '../VideoList';

describe('<VideoList />', () => {
  let videos;

  beforeEach(() => {
    videos = [
      {
        title: 'Test title',
        shortDescription: 'hello world',
        url: 'https://test.com/test',
      },
      {
        title: 'Test title 2',
        shortDescription: 'Lorem lipsum dolor sit amed',
        url: 'https://test.com/test2',
      },
    ];
  });

  it('should render list properly', () => {
    const wrapper = render(<VideoList videos={videos} />);

    expect(wrapper).toMatchSnapshot();
  });
});
