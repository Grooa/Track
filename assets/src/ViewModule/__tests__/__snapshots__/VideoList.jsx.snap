// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoList /> should render list properly 1`] = `
<ul
  class="course-list"
>
  <li>
    <a
      class="course-module clickable"
      href="https://test.com/test"
    >
      <div
        class="course-module-metadata"
      >
        <strong
          class="type"
        >
          Video 1
        </strong>
        <h3
          class="title"
        >
          Test title
        </h3>
      </div>
      <div
        class="description"
      >
        <p>
          hello world
        </p>
      </div>
      <button>
        Checkout
      </button>
    </a>
  </li>
  <li>
    <a
      class="course-module clickable"
      href="https://test.com/test2"
    >
      <div
        class="course-module-metadata"
      >
        <strong
          class="type"
        >
          Video 2
        </strong>
        <h3
          class="title"
        >
          Test title 2
        </h3>
      </div>
      <div
        class="description"
      >
        <p>
          Lorem lipsum dolor sit amed
        </p>
      </div>
      <button>
        Checkout
      </button>
    </a>
  </li>
</ul>
`;
